frontend/src/components/navigationSidebar/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import {
+  FaAngleDoubleLeft,
+  FaAngleDoubleRight,
+  FaChartLine,
+  FaList,
+  FaShare,
+  FaTrash,
+  FaUserFriends,
+} from 'react-icons/fa';
+import { AiOutlineFunction } from 'react-icons/ai';
+import { Role } from '../../utils/enums/roles';
+import {
+  NavigationSidebarToggleButton,
+  NavigationSidebarOuterWrapper,
+  NavigationSidebarInnerWrapper,
+  NavigationSidebarContentWrapper,
+} from './navigationSidebar-styles';
+import NavigationSidebarItem from './navigationSidebarItem';
+import { useAppSelector } from '../../store/hooks';
+import { RootState } from '../../store/store';
+import { Routes } from '../../utils/enums/routes';
+
+const NavigationSidebar = () => {
+  const user = useAppSelector((state: RootState) => state.user.user);
+  const [isOpen, setIsOpen] = useState(true);
+
+  const canAccessStatisticsPage =
+    user?.role === Role.ADMIN || user?.role === Role.MASTER_ADMIN;
+  const canAccessUserManagementPage = user?.role === Role.MASTER_ADMIN;
+
+  const handleNavigationSidebarToggle = () => {
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <NavigationSidebarOuterWrapper>
+      <NavigationSidebarInnerWrapper>
+        <NavigationSidebarToggleButton onClick={handleNavigationSidebarToggle}>
+          {isOpen ? <FaAngleDoubleLeft /> : <FaAngleDoubleRight />}
+        </NavigationSidebarToggleButton>
+        {isOpen && (
+          <NavigationSidebarContentWrapper>
+            <NavigationSidebarItem
+              label="Shelves"
+              icon={<FaList />}
+              to={Routes.DASHBOARD}
```

Comment:
what do you think about having some sort of object where you store data for sidebar links and just iterate through that object instead of writing code for every navbar item?

Reply:
I see what you think, it's a good idea to have a `links` array of objects with link props and map trough them. 
I'm just not sure what is the best way to handle the two links that are conditionally rendered based on user role. 
In this case I guess they can be added manually since there is only one of each, but if we had more links we could have something like `linksModeratorCanAccess` and `linksSuperAdminCanAccess` as well. Not sure, am I on the right track?

Reply:
Object can have , for example, which role can access..so when mapping , skip over links based on user role?